### ISSUE TITLE
[FEATURE] idea with @Builder or @Accessors  add Conditon or Function annotation, set field value with conditon

### ISSUE THREAD

**Issue by bijialin** [rzwitserloot/lombok#2470]
## problem
we use ``@Builder``  set entity field,sometimes we need set it with some conditon, just like object ``not null`` ,string ``not empty``..., if lombok can add a annotation to solve that   
### current  example
````
@Builder
public class User{
   private String name;
    private String mobile;
    private String email;
}

final UserBuilder builder = User.builder().name(name);
if(null!=mobile){
   builder.mobile(mobile);
}
if(null!=email){
   builder.email(email);
}
````

## solution
if lombok can create ``@Builder.Conditon`` annotation , to auto create method like this.
````
@Builder
public class User{
   private String name;

   @Builder.Conditon
    private String mobile;

    private String email;
}

# User class genarate code 

  public UserBuilder mobile(boolean condition, String mobile) {
    if (condition) {
      this.mobile = mobile;
    }
    return this;
  }

# use 
final UserBuilder builder = User.builder().name(name).mobile(null!=mobile,mobile);

```` 
when we use ``@Accessors`` set field value, it also same applies

**Comment by Maaartinus** [comment]
This feature probably wouldn't help much; the amount of code is about the same. Actually, in your example, the condition doesn't matter as `mobile` is `null` before the assignment.

In general, it could help to keep the fluent style.... however, I doubt it's worth it. Oftentimes, you could replace it by something like `.mobile(Strings.nullToEmpty(mobile))` or `.mobile(MoreObjects.firstNonNull(mobile, defaultMobile))` (both utils taken from Guava).

Disclaimer: I'm not a project owner.

**Comment by michaelboyles** [comment]
seems quite a specific use-case. consider instead

`Optional.ofNullable(mobile).ifPresent(builder::setMobile)`

**Comment by onacit** [comment]
> seems quite a specific use-case. consider instead
>
> `Optional.ofNullable(mobile).ifPresent(builder::setMobile)`

Good point. I searched issues to suggest a feature that a builder's invocation chain may use an `Optional` value.

Note that, using an `Optional` as a method parameter is not a good move, yet I see,

[public UriComponentsBuilder#queryParamIfPresent(String name, Optional value)](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/UriComponentsBuilder.html#queryParamIfPresent(java.lang.String,java.util.Optional))

which keeps invocation chaining available.

Can we generate additional setter methods take `Optional` which invoke original setter method when present?

```java
        public B userId(String userId) {
            this.userId = userId;
            return this.self();
        }

        public B userIdIfPresent(Optional<String> userId) {
            userId.ifPresent(this::userId);
            return this.self();
        }
```

```java
var userName = getUserName();
var userId = getUserId(); // Optional<String>
builder()
        .userName(userName)
        .userIdIfPresent(userId)
        .build();
```
